dbms/src/Storages/DeltaMerge/Index/LocalIndexInfo.cpp

```diff
@@ -82,6 +82,29 @@ ColumnID getVectorIndxColumnID(
     return EmptyColumnID;
 }
 
+void saveIndexFilePros(const LocalIndexInfo & index_info, dtpb::IndexFilePropsV2 * pb_idx, size_t uncompressed_size)
+{
+    switch (index_info.kind)
+    {
+    case TiDB::ColumnarIndexKind::Vector:
+    {
+        auto * pb_vec_idx = pb_idx->mutable_vector_index();
+        pb_vec_idx->set_format_version(0);
+        pb_vec_idx->set_dimensions(index_info.def_vector_index->dimension);
+        pb_vec_idx->set_distance_metric(tipb::VectorDistanceMetric_Name(index_info.def_vector_index->distance_metric));
```

Comment:
If we are going to refactor it, then we could refactor it more properly -- I think this should be assigned by individual IndexBuilders, not in a centerialized place.